build.sbt

```diff
@@ -386,21 +380,59 @@ lazy val typed = akkaModule("akka-typed")
   )
   .disablePlugins(MimaPlugin)
 
-lazy val typedTestkit = akkaModule("akka-typed-testkit")
-  .dependsOn(typed, testkit % "compile->compile;test->test")
-  .settings(AutomaticModuleName.settings("akka.typed.testkit"))
+lazy val persistenceTyped = akkaModule("akka-persistence-typed")
+  .dependsOn(
+    actorTyped,
+    persistence,
+    testkit % "test->test",
+    typedTestkit % "test->test",
+    actorTypedTests % "test->test"
+  )
+  .settings(AkkaBuild.mayChangeSettings)
+  .settings(AutomaticModuleName.settings("akka.persistence.typed"))
   .disablePlugins(MimaPlugin)
 
-lazy val typedTests = akkaModule("akka-typed-tests")
+lazy val clusterTyped = akkaModule("akka-cluster-typed")
   .dependsOn(
-    typed,
-    typedTestkit % "compile->compile;test->provided;test->test",
-    // the provided dependencies
-    persistence % "compile->compile;test->test",
-    cluster % "test->test",
+    actorTyped,
+    cluster,
     clusterTools,
+    distributedData,
+    persistence % "provided->test",
+    persistenceTyped % "provided->test",
```

Comment:
why dependencies to persistence from cluster?

Reply:
It is just for test, as we test having a persistent cluster singleton